src/wkt/src/internal.rs

```diff
@@ -17,73 +17,114 @@
 //! These types are intended for developers of the Google Cloud client libraries
 //! for Rust. They are undocumented and may change at any time.
 
-pub struct F32;
+pub type F32 = Float<f32>;
+pub type F64 = Float<f64>;
```

Comment:
I would like to avoid exporting a public type like `Float<T>`.  I wonder if we can keep F32 and F64 as unit types, something like:

```suggestion
pub type F32;
pub type F64;

trait FloatExt: num_traits::Float + std::fmt::Debug {
    type Type;
    fn serialize<S: serde::ser::Serializer>(self, serializer: S) -> Result<S::Ok, S::Error>;
}

impl FloatExt for f32 {
    type Type = f32;
    fn serialize<S: serde::ser::Serializer>(self, serializer: S) -> Result<S::Ok, S::Error> {
        serializer.serialize_f32(self)
    }
}

impl<T> serde_with::SerializeAs<T> for T where T: FloatExt + 'de {
  // Can reference T::Type here to get back the f32
```

As a last resort: using macros to generate the code below?

Reply:
Great suggestion, I'll give that a try.

I had left it as exported since it had to be in order to export the type aliases, but would much prefer to not.

Reply:
I fear I may have veered overly far in the macro  direction, but I think its better to avoid adding the num_traits dependency if necessary.

Might try another refactor to share the code without the macros.

Reply:
Seems like I'd need to bring back in generics, which I'm not yet convinced is better, so I think I'll take a review as is